### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by donggyu04**
Clean up

1. Deprecated.
2. Unused.
3. Indentation.
4. Redundant default parameter
5. IED(Intellij) warning.

### CODE REVIEW

File: ngrinder-controller/src/main/java/org/ngrinder/infra/config/WebMvcConfig.java [hunk 1]
```diff
@@ -37,14 +36,19 @@
...
-	@Autowired
-	private ResourceProperties resourceProperties = new ResourceProperties();
+	private final ResourceProperties resourceProperties;
+
+	public WebMvcConfig(LocaleChangeInterceptor localeChangeInterceptor,
+						DefaultSuccessJsonInterceptor defaultSuccessJsonInterceptor,
+						ResourceProperties resourceProperties) {
+		this.localeChangeInterceptor = localeChangeInterceptor;
+		this.defaultSuccessJsonInterceptor = defaultSuccessJsonInterceptor;
+		this.resourceProperties = resourceProperties;
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/infra/config/WebMvcConfig.java:50]: I highly recommend to use @RequiredArgsConstructor

Comment (donggyu04) [reply ngrinder-controller/src/main/java/org/ngrinder/infra/config/WebMvcConfig.java]: @junoyoon 

Fixed

File: ngrinder-controller/src/main/java/org/ngrinder/common/util/FileDownloadUtils.java [hunk 2]
```diff
@@ -59,12 +58,10 @@ public static boolean downloadFile(HttpServletResponse response, File file) {
...
 		response.addHeader("Content-Length", "" + file.length());
-		InputStream fis = null;
 		byte[] buffer = new byte[FILE_DOWNLOAD_BUFFER_SIZE];
-		OutputStream toClient = null;
-		try {
-			fis = new BufferedInputStream(new FileInputStream(file));
-			toClient = new BufferedOutputStream(response.getOutputStream());
+
+		try (InputStream fis = new BufferedInputStream(new FileInputStream(file));
+			 OutputStream toClient = new BufferedOutputStream(response.getOutputStream())) {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/common/util/FileDownloadUtils.java:64]: Simplify this by using ```IOUtils.copyLarge()```

File: ngrinder-controller/src/main/java/org/ngrinder/compatibility/perftest/controller/PerfTestFormTypeApiController.java [hunk 3]
```diff
@@ -35,8 +34,12 @@
 @Deprecated
 public class PerfTestFormTypeApiController extends PerfTestApiController {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/compatibility/perftest/controller/PerfTestFormTypeApiController.java:35]: @RequireArgsConstructor

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java [hunk 4]
```diff
@@ -762,25 +761,18 @@ public int getReportDataInterval(long testId, String dataType, int imgWidth) {
...
 
-		FileInputStream in = null;
-		InputStreamReader isr = null;
-		try {
-			in = new FileInputStream(targetFile);
-			isr = new InputStreamReader(in);
-			lnr = new LineNumberReader(isr);
+		try (FileInputStream in = new FileInputStream(targetFile);
+			 InputStreamReader isr = new InputStreamReader(in);
+			 LineNumberReader lnr = new LineNumberReader(isr)) {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java:767]: Simplify this

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java [hunk 5]
```diff
@@ -1172,25 +1164,19 @@ public int getMonitorGraphInterval(long testId, String targetIP, int imageWidth)
...
-		InputStreamReader isr = null;
-		LineNumberReader lnr = null;
 		int interval = 0;
-		try {
-			in = new FileInputStream(monitorDataFile);
-			isr = new InputStreamReader(in);
-			lnr = new LineNumberReader(isr);
+		try (FileInputStream in = new FileInputStream(monitorDataFile);
+			 InputStreamReader isr = new InputStreamReader(in);
+			 LineNumberReader lnr = new LineNumberReader(isr)) {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java:1170]: Simplify this

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java [hunk 6]
```diff
@@ -1336,26 +1321,20 @@ private File getReportPluginDataFile(Long testId, String plugin, String kind) {
...
-		InputStreamReader isr = null;
-		LineNumberReader lnr = null;
 		int interval = 0;
-		try {
-			in = new FileInputStream(dataFile);
-			isr = new InputStreamReader(in);
-			lnr = new LineNumberReader(isr);
+		try (FileInputStream in = new FileInputStream(dataFile);
+			 InputStreamReader isr = new InputStreamReader(in);
+			 LineNumberReader lnr = new LineNumberReader(isr)) {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java:1327]: Simplify this

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java [hunk 7]
```diff
@@ -1492,7 +1469,7 @@ public File getReportDataFile(long testId, String key) {
 
 		List<Float> metrics = new ArrayList<>();
 		try (FileReader reader = new FileReader(targetFile);
-			 BufferedReader br = new BufferedReader(reader);){
+			 BufferedReader br = new BufferedReader(reader)) {
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java:1472]: Simplify this

File: ngrinder-controller/src/main/java/org/ngrinder/perftest/service/samplinglistener/MonitorCollectorPlugin.java [hunk 8]
```diff
@@ -79,29 +78,26 @@ public void startSampling(final ISingleConsole singleConsole, PerfTest perfTest,
...
+				client.init();
+				if (client.isConnected()) {
+					File testReportDir = singleConsole.getReportPath();
+					File dataFile = null;
+					FileWriter fw = null;
+					BufferedWriter bw = null;
+					try {
+						dataFile = new File(testReportDir, MONITOR_FILE_PREFIX + target + ".data");
+						fw = new FileWriter(dataFile, false);
+						bw = new BufferedWriter(fw);
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/perftest/service/samplinglistener/MonitorCollectorPlugin.java:93]: Simplify this

File: ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java [hunk 9]
```diff
@@ -132,29 +130,25 @@ public File getUserRepoDirectory(User user) {
...
+						}
+						script.setPath(FilenameUtils.normalize(path + "/" + dirEntry.getRelativePath(), true));
+						script.setCreatedDate(dirEntry.getDate());
+						script.setLastModifiedDate(dirEntry.getDate());
+						script.setDescription(dirEntry.getCommitMessage());
+						script.setRevision(dirEntry.getRevision());
+						if (dirEntry.getKind() == SVNNodeKind.DIR) {
+							script.setFileType(FileType.DIR);
+						} else {
+							script.getFileType();
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java:147]: Check if this is necessary

File: ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java [hunk 10]
```diff
@@ -229,7 +221,6 @@ public FileEntry findOne(User user, String path, SVNRevision revision) {
 			if (nodeKind == SVNNodeKind.NONE) {
 				return null;
 			}
-			outputStream = new ByteArrayOutputStream();
 			SVNProperties fileProperty = new SVNProperties();
 			// Get File.
 			repo.getFile(path, revision.getNumber(), fileProperty, outputStream);
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java:226]: make ```try``` start here

File: ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java [hunk 11]
```diff
@@ -332,13 +320,12 @@ public void save(User user, FileEntry fileEntry, String encoding) {
...
-				if (fileEntry.getContentBytes() == null && fileEntry.getFileType().isEditable()) {
-					bais = new ByteArrayInputStream(checkNotNull(fileEntry.getContent()).getBytes(
-							encoding == null ? "UTF-8" : encoding));
-				} else {
-					bais = new ByteArrayInputStream(fileEntry.getContentBytes());
+
+				try (InputStream bais = fileEntry.getContentBytes() == null && fileEntry.getFileType().isEditable() ?
+					new ByteArrayInputStream(checkNotNull(fileEntry.getContent()).getBytes(encoding == null ? "UTF-8" : encoding)) :
+					new ByteArrayInputStream(fileEntry.getContentBytes())) {
+					checksum = deltaGenerator.sendDelta(fileEntry.getPath(), bais, editor, true);
```
Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java:327]: It's complex.. go back to old way

File: ngrinder-controller/src/test/java/org/ngrinder/agent/repository/AgentRepositoryTest.java [hunk 12]
```diff
@@ -19,9 +19,9 @@
 import org.ngrinder.model.AgentInfo;
 import org.springframework.beans.factory.annotation.Autowired;
 
+import static org.hamcrest.MatcherAssert.assertThat;
```
Comment (junoyoon) [inline ngrinder-controller/src/test/java/org/ngrinder/agent/repository/AgentRepositoryTest.java:22]: use then if possible in next pr.

File: ngrinder-controller/src/main/java/org/ngrinder/compatibility/perftest/controller/PerfTestFormTypeApiController.java (hunk 3 above)
Comment (donggyu04) [reply ngrinder-controller/src/main/java/org/ngrinder/compatibility/perftest/controller/PerfTestFormTypeApiController.java:35]: `PerfTestFormTypeApiController` is child class from  `PerfTestApiController`.

@RequireArgsConstructor can't call the parent constructor.

ref: https://stackoverflow.com/a/29771875/8074178
related issue: https://github.com/rzwitserloot/lombok/issues/375

File: ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java (hunk 10 above)
Comment (donggyu04) [reply ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java:226]: Above try block enclose SVNException, If move the block to here, we should have nested try-catch.

File: ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java (hunk 9 above)
Comment (donggyu04) [reply ngrinder-controller/src/main/java/org/ngrinder/script/repository/FileEntryRepository.java:147]: It sets fileType if null.

https://github.com/naver/ngrinder/blob/develop/ngrinder-controller/src/main/java/org/ngrinder/script/model/FileEntry.java#L112

**Review by imbyungjun — APPROVED** [review]

**Review by songeunwoo — APPROVED** [review]

**Comment by songeunwoo** [comment]
Thank you for your efforts.
The code looks good :)

**Pull request merged**